src/llvm-late-gc-lowering.cpp

```diff
@@ -2452,8 +2452,122 @@ void LateLowerGCFrame::PlaceRootsAndUpdateCalls(SmallVectorImpl<int> &Colors, St
     }
 }
 
+#ifdef MMTK_GC
```

Comment:
I'd prefer it if this was a runtime option to the pass rather than a compiletime option.

Reply:
Do you mean choosing MMTk as a runtime option or inlining the fastpath as a runtime option? The former is not really possible, at least not at this point. The latter should be possible, with maybe a small overhead of checking the runtime option variable every time we're running that pass.

Reply:
I mean the latter

Reply:
I have added an option `--inline-fastpath=no` that works when running Julia with MMTk and should disable inlining the code for fastpath allocation at runtime instead of at compile time.

Reply:
I think the idea is that the pass itself doesn't require changes at compile time. So that we can have tests over LLVM IR etc.

Reply:
We restructured the code splitting that pass using the `llvm-late-gc-lowering-stock.c` and `llvm-late-gc-lowering-mmtk.c` files. These files will get compiled according to which GC is used, but we also introduced the function [`jl_gc_active_impl`](https://github.com/mmtk/julia/blob/6d6077def9789535cb0d3e2765c9a9d245bdb3e6/src/gc-interface.h#L103), which should enable writing tests (over the LLVM IR, or in general) that are specific to each GC.